airbyte-integrations/connectors/source-relational-db/src/test/java/io/airbyte/integrations/source/relationaldb/StateDecoratingIteratorTest.java

```diff
@@ -147,6 +173,48 @@ void testCursorFieldIsEmpty() {
     assertFalse(iterator.hasNext());
   }
 
+  @Test
+  void testIteratorCatchesExceptionWhenEmissionFrequencyNonZero() {
+    final Iterator<AirbyteMessage> exceptionIterator = createExceptionIterator();
+    final StateDecoratingIterator iterator = new StateDecoratingIterator(
+        exceptionIterator,
+        stateManager,
+        NAME_NAMESPACE_PAIR,
+        UUID_FIELD_NAME,
+        RECORD_VALUE_1,
+        JsonSchemaPrimitive.STRING,
+        1);
+    assertEquals(RECORD_MESSAGE_1, iterator.next());
+    assertEquals(RECORD_MESSAGE_2, iterator.next());
+    // continues to emit RECORD_MESSAGE_2 since cursorField has not changed thus not satisfying the condition of "ready"
+    assertEquals(RECORD_MESSAGE_2, iterator.next());
+    assertEquals(RECORD_MESSAGE_3, iterator.next());
+    // emits the first state message since the iterator has changed cursorFields (2 -> 3) and met the frequency minimum of 1 record
+    assertEquals(STATE_MESSAGE_2, iterator.next());
+    // no further records to read since Exception was caught above and marked iterator as endOfData()
+    assertFalse(iterator.hasNext());
+  }
+
+  @Test
+  void testIteratorCatchesExceptionWhenEmissionFrequencyZero() {
+    final Iterator<AirbyteMessage> exceptionIterator = createExceptionIterator();
+    final StateDecoratingIterator iterator = new StateDecoratingIterator(
+        exceptionIterator,
+        stateManager,
+        NAME_NAMESPACE_PAIR,
+        UUID_FIELD_NAME,
+        RECORD_VALUE_1,
+        JsonSchemaPrimitive.STRING,
+        0);
+    assertEquals(RECORD_MESSAGE_1, iterator.next());
+    assertEquals(RECORD_MESSAGE_2, iterator.next());
+    assertEquals(RECORD_MESSAGE_2, iterator.next());
+    assertEquals(RECORD_MESSAGE_3, iterator.next());
+    // since stateEmission is not set to emit frequently, this will catch the error but not emit state message since it wasn't in a ready state
+    // of having a frequency > 0 but will prevent an exception from causing the iterator to fail by marking iterator as endOfData()
+    assertFalse(iterator.hasNext());
```

Comment:
Quick note: this may not be ideal since the cursor isn't updated when the `stateEmissionFrequency = 0` but ideally by having all connections use at least `1` or greater for `stateEmissionFrequency` this will emit state more frequently. The note here is that while this does not emit a state message, it does at least catch the exception and log that the messageIterator failed to read the next record